pkg/db/queries/queries.sql.go

```diff
@@ -150,6 +173,10 @@ type InsertGatewayEnvelopeParams struct {
 	OriginatorSequenceID int64
 	Topic                []byte
 	OriginatorEnvelope   []byte
+	BlockNumber          sql.NullInt64
+	BlockHash            []byte
```

Comment:
Just thinking out loud, but what if we did this relationally (like with a table called `block_versions`) and joined to that rather than denormalizing?

Would use less storage and still be easy to find impacted messages in a re-org.

Not a suggestion, more of a question.

Reply:
I made a similar proposal in slack :)

Reply:
I like that approach, too. My biggest concern is that for each group/identity message we'd be inserting in db twice and there would be additional storage overhead.

> Would use less storage
I'm not sure I follow, could you expand this?

Reply:
Just that we only have to store the block hash and number once per block, and store a foreign key on each message in the block. The block number and foreign key are the same size so they cancel out, but the block hash is pure savings and bigger.

I'm assuming most blocks have > 1 message in them.

Reply:
SQL schemas updated. It still needs some fine tuning, and better thinking on the queries, but captures the idea.